### PR TITLE
added filtering to getAllArticles

### PR DESCRIPTION
**Pull request by CallumBlejean**

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -130,6 +130,29 @@ describe("GET /api/articles", () => {
...
         });
     });
+    it("returns 200 and sorts articles by votes in ascending order with a valid topic", () => {
+      return request(app)
+      .get("/api/articles?sort_by=votes&order=asc&topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        articles.forEach((article) => {
+          console.log(article.topic)
```
Comment (LangaLee) [inline __tests__/api.test.js:140]: remove this log please

**Review by LangaLee — COMMENTED** [review]
- I would add a test for when a topic exists in the db but doesn't have any articles under it such as the topic "paper"

File: __tests__/api.test.js (hunk 1 above)
Comment (CallumBlejean) [reply __tests__/api.test.js:140]: it was removed in the next push

**Comment by CallumBlejean** [comment]
> * I would add a test for when a topic exists in the db but doesn't have any articles under it such as the topic "paper"

pushing this change now

**Pull request merged**